src/MetadataUtility/Metadata/SupportFiles/FrontierLabs/LogFile.cs

```diff
@@ -39,146 +50,339 @@ public static void FindLogFile(TargetInformation information, IEnumerable<string
+                EraseBlSize = uint.Parse(reader.ReadLine()!.Split().Last()),
+            };
+        }
+
+        /// <summary>
+        /// Parses a location from a log file.
+        /// </summary>
+        /// <param name="value">The unparsed location value.</param>
+        /// <returns>
+        /// A parsed location object.
+        /// </returns>
+        public static Location LocationParser(string value)
+        {
+            double? longitude, latitude;
+
+            value = NumericParser(value);
+
+            // Find index dividing lat and lon
+            int latLonDividingIndex = value.IndexOfAny(new char[] { '+', '-' }, 1);
+
+            // Parse lat and lon
+            latitude = double.Parse(value.Substring(0, latLonDividingIndex));
+            longitude = double.Parse(value.Substring(latLonDividingIndex));
+
+            return new Location() with
+            {
+                Latitude = latitude,
+                Longitude = longitude,
+            };
+        }
+
+        /// <summary>
+        /// Parses battery data from a log file.
+        /// </summary>
+        /// <param name="value">The unparsed battery data.</param>
+        /// <returns>
+        /// Parsed battery data.
+        /// </returns>
+        public static (double? BatteryLevel, double? Voltage) BatteryParser(string value)
+        {
+            string[] batteryValues = value.Split(" ", StringSplitOptions.RemoveEmptyEntries);
+
+            double? batteryLevel = double.Parse(NumericParser(batteryValues[0])) / 100;
+            double? batteryVoltage = double.Parse(NumericParser(batteryValues[1]));
+
+            return (batteryLevel, batteryVoltage);
+        }
+
+        /// <summary>
+        /// Parses a microphone from a log file.
+        /// </summary>
+        /// <param name="value">The unparsed microphone value.</param>
+        /// <returns>
+        /// A parsed microphone object.
+        /// </returns>
+        public static Microphone MicrophoneParser(string value)
+        {
+            if (value.Contains("Unknown"))
+            {
+                return null;
+            }
+
+            char? channelName = null;
+            int? channel = null;
+
+            // Parse channel data if it exists
+            if (value.Contains("Ch"))
+            {
+                channelName = value.Split(':').First().ToArray().Last();
+                channel = (int)channelName - 64;
             }
 
-            return logFile;
+            // Split string into individual microphone values
+            string[] micValues = value.Split("\"").Select(x => x.Trim()).ToArray();
+
+            // Parse each value
+            string uid = micValues[0].Split(" ").Last();
+            string type = micValues[1];
+            string buildDate = string.Join(
+                '-',
+                new string(micValues[2].Where(c => !(new char[] { ' ', '(', ')' }).Contains(c)).ToArray())
+                .Split('/').Reverse().ToArray());
```

Comment:
This is pretty messy... in the log file mic manufacture date is given as dd/mm/yyyy, rather than what we have had elsewhere (yyyy-mm-dd). Maybe makes more sense for this to be Nodatime, what do you think?

Reply:
This seems like a good job for a regular expression to extract the date, and then yes, add a NodaTime format parser for that date variant.